software/25.Flutter/README.md

```diff
@@ -0,0 +1,195 @@
+      debugShowCheckedModeBanner: false,
+    );
+  }
+}
+```
+
+Create a folder `lib/pages` and put a `home.dart` file with this code in it:
+```dart
+import 'package:flutter/material.dart';
+
+class MyHomePage extends StatefulWidget {
+  const MyHomePage({Key? key}) : super(key: key);
+
+  @override
+  State<MyHomePage> createState() => _MyHomePageState();
+}
+
+class _MyHomePageState extends State<MyHomePage> {
+  @override
+  Widget build(BuildContext context) {
+    return const Scaffold(
+      body: SafeArea(
+        child: Center(
+          child: Text("Step 1"),
+        ),
+      ),
+    );
+  }
+}
+```
+
+If everything has worked and you launche the project, you migth have the text `Step 1` in the middle of the application.
+
+:warning: In this code you have some const that you will have to remove because of non const contructor of widget you will use.
+
+## Step 2 - Lists
+
+The most important thing in a `to do list` is to have a list of `Widget` that will represent your todos.
+
+For this step you have to create a folder `lib/components` and put a `todo_list.dart` file in it. You can now past this code in the file:
+
+```dart
+import 'package:flutter/material.dart';
+
+class ToDoList extends StatefulWidget {
+  const ToDoList({Key? key, required this.arr}) : super(key: key);
+
+  final List<String> arr;
+  @override
+  State<ToDoList> createState() => _ToDoListState();
+}
+
+class _ToDoListState extends State<ToDoList> {
+  @override
+  Widget build(BuildContext context) {
+    // implement list instead of the code below.
+    return Center(
+      child: Text("To do list"), 
+    );
+  }
+}
+```
+
+This code create a `component` `ToDoList` that take a list of string in parameters.
+
+Here is to know [how to access of parameters](https://stackoverflow.com/questions/50287995/passing-data-to-statefulwidget-and-accessing-it-in-its-state-in-flutter).
+
+Lets take a look of how to do list with Flutter. For it you have muliple choices, here are few of them:
+
+- [List.generate](https://api.flutter.dev/flutter/dart-core/List/List.generate.html)
+
+
+    As there is no exemple with `widget` in the doc, here is one: 
+    
+    ```dart
+    Column(
+        children: List.generate(
+            10,
+            (index) {
+                return Text(index.toString());
+            }
+        ),
+    )
+    ```
+- [ListView.builder](https://docs.flutter.dev/cookbook/lists/long-lists)
+
+- [For loop](https://stackoverflow.com/questions/56947046/flutter-for-loop-to-generate-list-of-widgets)
+
+You have to make sure you component is workink by calling your component with this list in parameters:
+
+```dart
+final List<String> _arr = ["Todo 1", "Todo 2", "Todo 3", "Todo 4", "Todo 5"];
+```
+
+:warning: If the list can't be display to the entire screen (it will happen), you might wrap your list with [SingleChildScrollView](https://api.flutter.dev/flutter/widgets/SingleChildScrollView-class.html), to make your screen scrollable.
```

Comment:
You need to define what is a widget and what is it utility. Also you need to more precise where we need to code (for example I dont't know where to call my to do list). You must not consider that the reader has experience or knowledge with Flutter. You must start from zero you need to explane everything or to redirect to good documentation. This comment is available for the others steps.